contracts/src/v0.8/shared/test/util/SortedSetValidationUtil.t.sol

```diff
@@ -0,0 +1,168 @@
+    uint256 supersetLength
+  ) internal pure returns (bytes32[] memory subset, bytes32[] memory superset) {
+    subset = new bytes32[](subsetLength);
+    superset = new bytes32[](supersetLength);
+  }
+
+  function _convertArrayToSortedSet(bytes32[] memory arr, uint256 offSet) internal pure {
+    for (uint256 i = 1; i < arr.length; ++i) {
+      arr[i] = bytes32(uint256(arr[i - 1]) + offSet);
+    }
+  }
+
+  function _convertArrayToUnsortedSet(bytes32[] memory arr, uint256 offSet, uint256 ptr1, uint256 ptr2) internal pure {
+    _convertArrayToSortedSet(arr, offSet);
+    // Swap two elements to make it unsorted
+    (arr[ptr1], arr[ptr2]) = (arr[ptr2], arr[ptr1]);
+  }
+
+  function _convertArrayToSubset(bytes32[] memory subset, bytes32[] memory superset) internal pure {
+    for (uint256 i; i < subset.length; ++i) {
+      subset[i] = superset[i];
+    }
+  }
+
+  function _makeInvalidSubset(bytes32[] memory subset, bytes32[] memory superset, uint256 ptr) internal pure {
+    _convertArrayToSubset(subset, superset);
+    subset[ptr] = bytes32(uint256(subset[ptr]) + 1);
+  }
+
+  function _convertArrayToHaveDuplicates(bytes32[] memory arr, uint256 ptr1, uint256 ptr2) internal pure {
+    arr[ptr2] = arr[ptr1];
+  }
+}
+
+contract SortedSetValidationUtil_CheckIsValidUniqueSubsetTest is SortedSetValidationUtilBaseTest {
+  // Successes.
+
+  function test__checkIsValidUniqueSubset_ValidSubset_Success() public {
+    (bytes32[] memory subset, bytes32[] memory superset) = _createSets(3, 5);
+    _convertArrayToSortedSet(superset, OFFSET);
+    _convertArrayToSubset(subset, superset);
+
+    SortedSetValidationUtil._checkIsValidUniqueSubset(subset, superset);
+  }
+
+  // Reverts.
+
+  function test__checkIsValidUniqueSubset_EmptySubset_Reverts() public {
+    (bytes32[] memory subset, bytes32[] memory superset) = _createSets(0, 5);
+    _convertArrayToSortedSet(superset, OFFSET);
+
+    vm.expectRevert(abi.encodeWithSelector(SortedSetValidationUtil.EmptySet.selector));
+    SortedSetValidationUtil._checkIsValidUniqueSubset(subset, superset);
+  }
+
+  function test__checkIsValidUniqueSubset_EmptySuperset_Reverts() public {
+    (bytes32[] memory subset, bytes32[] memory superset) = _createSets(3, 0);
+    _convertArrayToSortedSet(subset, OFFSET);
+
+    vm.expectRevert(abi.encodeWithSelector(SortedSetValidationUtil.EmptySet.selector));
+    SortedSetValidationUtil._checkIsValidUniqueSubset(subset, superset);
+  }
+
+  function test__checkIsValidUniqueSubset_NotASubset_Reverts() public {
+    (bytes32[] memory subset, bytes32[] memory superset) = _createSets(3, 5);
+    _convertArrayToSortedSet(superset, OFFSET);
+    _makeInvalidSubset(subset, superset, 1);
+
+    vm.expectRevert(abi.encodeWithSelector(SortedSetValidationUtil.NotASubset.selector, subset, superset));
+    SortedSetValidationUtil._checkIsValidUniqueSubset(subset, superset);
+  }
+
+  function test__checkIsValidUniqueSubset_NotASortedSet_Reverts() public {
```

Comment:
nit: for consistency we should use the same naming pattern `NotASortedSubset` / `NotASortedSuperset` or `UnsortedSuperSet` / `UnsortedSubset`

Reply:
https://github.com/smartcontractkit/chainlink/pull/13774/commits/2146c8516c1c097b22e1a1e6a0f8400529f0b0f8